flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -566,6 +619,67 @@ public void setInitialState(@Nullable JobManagerTaskRestore taskRestore) {
 		}
 	}
 
+	@VisibleForTesting
+	CompletableFuture<Execution> registerProducedPartitions(TaskManagerLocation location) {
+		assertRunningInJobMasterMainThread();
+
+		return FutureUtils.thenApplyAsyncIfNotDone(
+			registerProducedPartitions(vertex, location, attemptId),
+			vertex.getExecutionGraph().getJobMasterMainThreadExecutor(),
+			producedPartitionsCache -> {
+				producedPartitions = producedPartitionsCache;
+				return this;
+			});
+	}
+
+	@VisibleForTesting
+	static CompletableFuture<Map<IntermediateResultPartitionID, ResultPartitionDeploymentDescriptor>> registerProducedPartitions(
```

Comment:
why static here

Reply:
why not? it is also easier to test without Execution mocking.